project2-beat-that/script.js

```diff
@@ -1,4 +1,124 @@
+  // return 5; - testing draw results
+  var randomDecimal = Math.random() * 6;
+  var randomInteger = Math.floor(randomDecimal);
+  var diceNumber = randomInteger + 1;
+  return diceNumber;
+};
+
+// global variables
+// initial game mode
+var gameMode = "playerOneRoll";
+// arrays to store the dice numbers rolled
+var playerOneNumbers = [];
+var playerTwoNumbers = [];
+// for the player combined numbers
+var playerOneScore;
+var playerTwoScore;
+// player accumulated score
+//var totalPlayerOneScore;
+//var totalPlayerTwoScore;
+
+// game results helper function - can it be here??
+var gameResults = function () {
+  var winnerMessage = `Player One's Number: ${playerOneScore} <br> Player Two's Number: ${playerTwoScore}`;
+  var winnerScore = Math.min(playerOneScore, playerTwoScore);
+  console.log(playerOneScore, playerTwoScore, winnerScore);
+  if (playerOneScore == playerTwoScore) {
+    winnerMessage += "<br> It's a draw!";
+  } else if (playerOneScore == winnerScore) {
+    winnerMessage += "<br> Player 2 Wins! 🏆";
+  } else winnerMessage += "<br> Player 1 Wins! 🏆";
+  return winnerMessage;
+};
+
+// main function
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  // initial myOutputValue
+  var myOutputValue = "";
+
+  // in the initial game mode, player 1 to roll the dice twice
+  if (gameMode === "playerOneRoll") {
+    var rollDice1 = rollDice();
+    var rollDice2 = rollDice();
+    // to keep record of the dice numbers
+    playerOneNumbers[0] = rollDice1;
+    playerOneNumbers[1] = rollDice2;
+    // output dice number rolled and instruction for next game step
+    myOutputValue = `Player One has rolled ${playerOneNumbers[0]} and ${playerOneNumbers[1]}. <br> Please choose order 1 or 2.`;
+    // change to the next game mode
+    gameMode = "playerOneOrder";
+  }
+  // in the next game mode, player 1 to choose the order of their dice numbers
+  else if (gameMode === "playerOneOrder") {
+    console.log(gameMode);
+    // input validation
+    if (Number(input) !== 1 && Number(input) !== 2) {
+      myOutputValue = `Player One, you rolled ${playerOneNumbers[0]} and ${playerOneNumbers[1]}. <br> Please enter a valid order choice - 1 or 2`;
+      return myOutputValue;
+    }
+    // input order choice of 1 or 2
+    if (Number(input) === 1) {
+      playerOneScore = Number(
+        String(playerOneNumbers[0]) + String(playerOneNumbers[1])
+      );
+      myOutputValue = `Player One, you have chosen your number as ${playerOneScore}. <br> Player Two can roll now.`;
+      // change the game mode to player 2's turn to roll
+      gameMode = "playerTwoRoll";
+    }
+    if (Number(input) === 2) {
+      playerOneScore = Number(
+        String(playerOneNumbers[1]) + String(playerOneNumbers[0])
+      );
+      myOutputValue = `Player One, you have chosen your number as ${playerOneScore}. <br> Player Two can roll now.`;
+      // change the game mode to player 2's turn to roll
+      gameMode = "playerTwoRoll";
+    }
```

Comment:
- An if-else if block should be better here since they are testing the same variable.
- Good way of concatenating the numbers together here - it shows that you understand the differences in data types :) 
- Note that lines 66-68 and 74-76 are the same, you can take them out of the if statements and just put it below before the game mode ends. That way it'll run in both situations, and you reduce the amount of repeated code you have.